sympy/ntheory/continued_fraction.py

```diff
@@ -158,9 +256,14 @@ def untillist(cf):
         solns = solve(continued_fraction_reduce(period + [y]) - y, y)
         solns.sort()
         pure = solns[-1]
-        return a.subs(x, pure).radsimp()
+        rv = a.subs(x, pure).radsimp()
     else:
-        return a
+        rv = a
+    if rv.is_Add:
+        rv = factor_terms(rv)
+        if rv.is_Mul and rv.args[0] == -1:
+            rv = rv.func(*rv.args)
```

Comment:
Will this modify `rv` in some way?

Reply:
Yes, it will evaluate it and distribute the minus sign if it looks like `-(1+sqrt(2))`.

Reply:
Hmm.. That is curious.
```
>>> rv = factor_terms(-(1 + sqrt(2)))
>>> rv == rv.func(*rv.args)
False
```
However, the [documents](https://docs.sympy.org/latest/tutorial/manipulation.html#args) say:
> Every well-formed SymPy expression must either have empty args or satisfy expr == expr.func(*expr.args).

It looks like the result of `factor_terms` is not strictly "well-formed".

Reply:
Yes, it is an unevaluated expression and that behavior is expected for them, e.g. p = Pow(1, 2, evaluate=False); p != p.func(*p.args)`